libs/d3d12/main.c

```diff
@@ -233,39 +197,46 @@ static VkPhysicalDevice d3d12_get_vk_physical_device(struct vkd3d_instance *inst
     if ((vr = pfn_vkEnumeratePhysicalDevices(vk_instance, &count, vk_physical_devices)) < 0)
         goto done;
 
-    if (pfn_vkGetPhysicalDeviceProperties2)
+    TRACE("Matching adapters by LUIDs.\n");
+
+    for (i = 0; i < count; ++i)
     {
-        TRACE("Matching adapters by LUIDs.\n");
+        pfn_vkGetPhysicalDeviceProperties(vk_physical_devices[i], &properties2.properties);
 
-        for (i = 0; i < count; ++i)
-        {
-            memset(&id_properties, 0, sizeof(id_properties));
-            id_properties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ID_PROPERTIES;
+        /* Skip over < 1.1 physical devices because... 1.1 instances
+         * still expose those apparently... */
+        if (properties2.properties.apiVersion < VK_API_VERSION_1_1)
+            continue;
 
-            properties2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2;
-            properties2.pNext = &id_properties;
+        memset(&id_properties, 0, sizeof(id_properties));
+        id_properties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ID_PROPERTIES;
 
-            pfn_vkGetPhysicalDeviceProperties2(vk_physical_devices[i], &properties2);
+        properties2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2;
+        properties2.pNext = &id_properties;
 
-            if (!memcmp(id_properties.deviceLUID, &adapter_desc->AdapterLuid, VK_LUID_SIZE))
-            {
-                vk_physical_device = vk_physical_devices[i];
-                break;
-            }
+        pfn_vkGetPhysicalDeviceProperties2(vk_physical_devices[i], &properties2);
+
+        if (!memcmp(id_properties.deviceLUID, &adapter_desc->AdapterLuid, VK_LUID_SIZE))
```

Comment:
You should check `id_properties.deviceLUIDValid` first.

Reply:
Resolved.